lib/node_modules/@stdlib/string/len/test/test.js

```diff
@@ -0,0 +1,269 @@
+	t.strictEqual( out, V, 'returns break property' );
+
+	out = getGraphemeBreakProperty( 0x11FE );
+	t.strictEqual( out, T, 'returns break property' );
+
+	out = getGraphemeBreakProperty( 0xAEBC );
+	t.strictEqual( out, LV, 'returns break property' );
+
+	out = getGraphemeBreakProperty( 0xAFD2 );
+	t.strictEqual( out, LVT, 'returns break property' );
+
+	out = getGraphemeBreakProperty( 0x200D );
+	t.strictEqual( out, ZWJ, 'returns break property' );
+
+	out = getGraphemeBreakProperty( 0x000A );
+	t.strictEqual( out, LF, 'returns break property' );
+
+	out = getGraphemeBreakProperty( 0x000D );
+	t.strictEqual( out, CR, 'returns break property' );
+
+	t.end();
+});
+
+tape( 'the getEmojiProperty function returns emoji property', function test( t ) {
+	var out;
+
+	out = getEmojiProperty( 0x23EC );
+	t.strictEqual( out, ExtendedPictographic, 'returns Extended_Pictographic' );
+
+	out = getEmojiProperty( 0x1FFFE );
+	t.strictEqual( out, Other, 'returns Other' );
+
+	t.end();
+});
+
+tape( 'the isSurrogate function returns whether string has surrogate pair', function test( t ) {
+	var out;
+
+	out = isSurrogate( 'अनुच्छेद', 2 );
+	t.strictEqual( out, false, 'returns false' );
+
+	out = isSurrogate( '🌷', 0 );
+	t.strictEqual( out, true, 'returns true' );
+
+	t.end();
+});
+
+tape( 'the nextBreak function returns the next grapheme break position in the string', function test( t ) {
+	var out;
+
+	out = nextBreak( 'last man standing', 4 );
+	t.strictEqual( out, 5, 'returns next grapheme break' );
+
+	out = nextBreak( 'अनुच्छेद', 1 );
+	t.strictEqual( out, 3, 'returns next grapheme break' );
+
+	out = nextBreak( '🌷', 0 );
+	t.strictEqual( out, 2, 'returns next grapheme break' );
+
+	t.end();
+});
+
+tape( 'the main function throws an error if not provided a string', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		5,
+		null,
+		true,
+		undefined,
+		NaN,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			len( value );
+		};
+	}
+});
+
+tape( 'the main function returns 0 if provided an empty string', function test( t ) {
+	t.strictEqual( len( '' ), 0, 'returns 0 if provided an empty string' );
+	t.end();
+});
+
+tape( 'the main function returns the Unicode aware length of a given string', function test( t ) {
+	var out;
+
+	out = len( 'hello world' );
+	t.strictEqual( out, 11, 'returns length' );
+
+	out = len( '!!!' );
+	t.strictEqual( out, 3, 'returns length' );
+
+	out = len( 'अनुच्छेद' );
+	t.strictEqual( out, 5, 'returns Unicode aware length' );
+
+	out = len( '🌷' );
+	t.strictEqual( out, 1, 'returns Unicode aware length' );
+
+	t.end();
+});
+
+tape( 'the main function returns the Unicode aware length for official Unicode 13 testcases', function test( t ) {
+	var testData;
+	var out;
+	var i;
+
+	function ucs2encode( array ) {
```

Comment:
Instead of dynamically generating the fixture data upon running tests, may be good to pre-compile the fixture data as JSON and store alongside the `*.txt` file in the `test/fixtures/*` folder. This will ensure that the tests can be run in the browser. Currently, they cannot.

Accordingly, may be worthwhile moving the data processing logic to a `test/fixtures/script.js` file which loads the fixture data and then writes to a JSON file. If we ever update the `*.txt` file, we'd need to rerun this test fixture "runner" in order to ensure that the fixture data remains in sync. We follow a similar process in, e.g., our base special mathematical functions tests, so this approach already has precedent in the project.